core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -1364,6 +1369,10 @@ public int getLostBlocksCount() {
     return mLostBlocks.size();
   }
 
+  private int getToRemoveBlockCount() {
+    return mWorkers.stream().map(MasterWorkerInfo::getToRemoveBlockCount).reduce(0, Integer::sum);
```

Comment:
Proper locking is required. You can follow the patter in `getCapacityBytes()`. Check the `MasterWorkerInfo` java doc too.

Reply:
Locking should not hurt your performance as long as you use the cached gauge.